### ISSUE TITLE
qt 5.9.3 static version doesn't build against Linuxbrew's zlib

### ISSUE THREAD

**Issue by fcying** [Linuxbrew/homebrew-core#5345]
# Please fill out the issue checklist below, and provide all of the requested information.

# Please always follow these steps:
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Linuxbrew/brew/issues/new.
- [ ] If you have a macOS system and can replicate the problem on macOS with Homebrew, please report the issue to [Homebrew/core](https://github.com/Homebrew/homebrew-core) and follow their issue template from your macOS system. If you are unsure, please report the issue to Linuxbrew.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [ ] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
I download the offical qt 5.9.3, and want to build a static version.
- What happened (include command output)
build error
```
qt/5.9.3/Src/qtbase/lib/libqtlibpng.a(pngrutil.o):pngrutil.c:function png_inflate_claim: error: undefined reference to 'inflateValidate'
collect2: error: ld returned 1 exit status
make[3]: *** [../../bin/qmlprofiler] Error 1
```
if I run `brew uninstall  --ignore-dependencies zlib`, and use os(ubuntu16.04) default zlib, it build fine
- What you expected to happen
build ok
- Step-by-step reproduction instructions (by running `brew install` commands)
```
cd ~/qt/5.9.3/Src
./configure -prefix ~/qt-5.9.3 -static -release -nomake examples
make 
```

**Comment by iMichka** [comment]
Hi. I is unclear to me if you are building breewed qt or trying to build qt from source by yourself.

Also, the error message you are getting is about libpng.

**Comment by fcying** [comment]
I'm trying to build qt from source by myself.

I try `brew install libpng or libpng@1.2`, build failed also.

maybe brew's zlib version against qt's libqtlibpng.a?

**Comment by iMichka** [comment]
We do provide a brew `qt`, why not use that? We did not add a static build to the formula though. So this is completely untested.

I guess that ./configure may be mixing the wrong png/zlib libraries. You need to read the `qt` doc to find out how to pass the right paths to ./configure. Probably just by setting the `-I` and `-L` flags correctly.

**Comment by fcying** [comment]
brew qt can not installed, it dependenci berkeley-db, and python2.7 need berkeley-db@4
```
brew install qt
Error: qt contains conflicting version recursive dependencies:
  berkeley-db@4, berkeley-db  

brew uninstall berkeley-db@4
Error: Refusing to uninstall .linuxbrew/Cellar/berkeley-db@4/4.8.30
because it is required by python 2.7.14, which is currently installed.
```

I make qt static version by use `-L` to link system's zlib, build ok.

Can brew support another zlib version for selection? I saw the system zlib version is 1.2.8, brew zlib version is 1.2.11

**Comment by iMichka** [comment]
For `qt`, this is a new bug: I opened a PR for that here: #5378

> Can brew support another zlib version for selection?

Not directly. You would have to write a `zlib@1.2.8` formula. We prefer that everybody uses the newest version.

**Comment by fcying** [comment]
Thanks, I will learn to write a formula.

**Comment by iMichka** [comment]
The qt dependency problem has been fixed with #5386.

Why would you need an older zlib version? It is older and has probably more bugs. I do not recommend doing this (even if it is technically possible to write a formula for this). Please note that we will not accept that formula in Linuxbrew.

**Comment by fcying** [comment]
I update to the laster brew version.
modify `qt.rb` : add `-static` and remove `RPATH`, it can build static version, not need an older zlib version now,  thanks~
